examples/timeseries/timeseries_classification_from_scratch.py

```diff
@@ -0,0 +1,231 @@
+"""
+"""
+## Introduction
+
+This example shows how to do time series classification from scratch, starting from raw
+CSV time series files on disk, without leveraging pre-trained weights or a pre-made Keras
+Application model. We demonstrate the workflow on Coffee dataset from the [UCR/UEA
+dataset](https://www.cs.ucr.edu/%7Eeamonn/time_series_data_2018/).
+
+"""
+
+"""
+## Setup
+
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn import preprocessing
+
+"""
+## Load the data: the Coffee dataset
+The description of this dataset can be found
+[here](https://archive.uea.ac.uk/~ajb/Papers/BagnallSDM2012.pdf).
+
+### Read the TSV data online
+We will use the `Coffee_TRAIN` file for training and the
+`Coffee_TEST` file for testing. The simplicity of this dataset
+allows us to demonstrate effectively how to use ConvNets for time series classification.
+In this file, the first column corresponds to the label.
+"""
+
+
+def readucr(filename):
+    data = np.loadtxt(filename, delimiter="\t")
+    Y = data[:, 0]
+    X = data[:, 1:]
+    return X, Y
+
+
+root_url = "https://germain-forestier.info/dataset/UCRArchive_2018/Coffee/"
+
+x_train, y_train = readucr(root_url + "Coffee_TRAIN")
+x_test, y_test = readucr(root_url + "Coffee_TEST")
+
+"""
+## Visualize the data
+Here we visualize one time series example for each class in the dataset.
+
+"""
+
+classes = np.unique(np.concatenate((y_train, y_test), axis=0))
+
+plt.figure()
+for c in classes:
+    c_x_train = x_train[y_train == c]
+    plt.plot(c_x_train[0], label="class" + str(c))
+plt.legend(loc="best")
+plt.show()
+plt.close()
+
+"""
+## Standardizing the data
+
+Our time series are already in a single length (176). However, their values are
+usually in various ranges. This is not ideal for a neural network;
+in general we should seek to make the input values normalized. Here, we will
+standardize values to have a mean equal to zero and a standard deviation equal to one.
+We perform this step for each train and test set.
+"""
+
+std_ = x_train.std(axis=1, keepdims=True)
+std_[std_ == 0] = 1.0
+x_train = (x_train - x_train.mean(axis=1, keepdims=True)) / std_
+
+std_ = x_test.std(axis=1, keepdims=True)
+std_[std_ == 0] = 1.0
+x_test = (x_test - x_test.mean(axis=1, keepdims=True)) / std_
+
+"""
+Note that the time series data used here are univariate, meaning we only have one channel
+per time series example.
+We will therefore transform the time series into a multivariate one with one channel
+using a simple reshaping via numpy.
+This will allow us to construct a model that is easily applicable to multivariate time
+series.
+"""
+
+x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], 1))
+x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], 1))
+
+"""
+Finally, in order to have a standard labeling of the data we will use the enconding from
```

Comment:
You don't need to. You can use sparse_categorical_crossentropy instead. Just count how many categories you have in the data beforehand